src/IntrusivePtr.h

```diff
@@ -0,0 +1,216 @@
+// See the file "COPYING" in the main distribution directory for copyright.
+
+#pragma once
+
+#include <type_traits>
+#include <utility>
+
+// An intrusive, reference counting smart pointer implementation.
```

Comment:
Mind adding some more API docs for this class? Given that it's now hardcoded to Ref/Unref, it'd be good to describe things from that angle: how is this pointer used to ref'cnt `BroObj` instances? A summary for the class would be good, and some more detail at least on methods that's aren't completely obvious. Basically, make it easy/obvious for people to use this class going forward.